From d548f53eb7a79a244f458d7d4c306defe82ab1b7 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Tue, 21 May 2013 12:23:05 +1000
Subject: [PATCH 16/16] Don't overwrite the cookies serial number

serial != sequenceNumber, see _XSetLastRequestRead()

cookie->serial is already set at this point, setting it again directly from
the sequenceNumber of the event causes a bunch of weird issues such as
scrollbars and text drag-n-drop breaking.

https://bugzilla.redhat.com/show_bug.cgi?id=965347

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
(cherry picked from commit 661c45ca17c434dbd342a46fd3fb813852ae0ca9)
(cherry picked from commit 49307ef3e8b4a24b501729f0a9551d10cc53b9c5)
---
 src/XExtInt.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/src/XExtInt.c b/src/XExtInt.c
index c7e6128..5fc7afc 100644
--- a/src/XExtInt.c
+++ b/src/XExtInt.c
@@ -891,7 +891,6 @@ static void xge_copy_to_cookie(xGenericEvent* ev,
     cookie->type = ev->type;
     cookie->evtype = ev->evtype;
     cookie->extension = ev->extension;
-    cookie->serial = ev->sequenceNumber;
 }
 
 static Bool
-- 
1.7.7.1

